File: hw/calle/u3/hw29Singleton/JAVA/src/main/java/ec/edu/espe/monthlyrentmanager/model/Resident.java
package ec.edu.espe.monthlyrentmanager.model;

import ec.edu.espe.monthlyrentmanager.utils.RentalManager;

/**
 *
 * @author T.A.P, The Art of Programming, @ESPE
 */
public class Resident {
    private int id;
    private String name;
    private int months;

    public Resident(int id, String name, int months) {
        this.id = id;
        this.name = name;
        this.months = months;
    }

    public double calculateTotalRent() {
        double monthlyRent = RentalManager.getInstance().getMonthlyRent();
        return monthlyRent * months;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public int getMonths() {
        return months;
    }
}

File: hw/aymacana/u3/hw29Singleton/SingletonPython/model/resident.py
# model/resident.py
from utils.rental_manager import RentalManager

class Resident:
    def __init__(self, id: int, name: str, months: int):
        self.id = id
        self.name = name
        self.months = months
    
    def calculate_total_rent(self) -> float:
        monthly_rent = RentalManager.get_instance().get_monthly_rent()
        return monthly_rent * self.months
    
    def get_id(self) -> int:
        return self.id
    
    def get_name(self) -> str:
        return self.name
    
    def get_months(self) -> int:
        return self.months